src/sidebar/reducers/annotations.js

```diff
@@ -173,17 +173,16 @@ var update = {
 
   UPDATE_ANCHOR_STATUS: function (state, action) {
     var annotations = state.annotations.map(function (annot) {
-      var match = (annot.id && annot.id === action.id) ||
-                  (annot.$tag && annot.$tag === action.tag);
-      if (match) {
-        return Object.assign({}, annot, {
-          $anchorTimeout: action.anchorTimeout || annot.$anchorTimeout,
-          $orphan: action.isOrphan,
-          $tag: action.tag,
-        });
-      } else {
+      if (!action.statusUpdates.hasOwnProperty(annot.$tag)) {
         return annot;
       }
+
+      var state = action.statusUpdates[annot.$tag];
+      if (state === 'timeout') {
+        return Object.assign({}, annot, { $anchorTimeout: true });
+      } else {
+        return Object.assign({}, annot, { $orphan: state === 'orphan' });
+      }
```

Comment:
And for each annotation that gets updated, we _either_ set `$anchorTimeout: true` _or_ we set `$orphan` to either `true` or `false`. Here I'm confused. Isn't "anchor timeout" and "orphan" kind of the same thing? And I'm not sure why this PR is changing this block of code, from the `if (match)` block that was here before.

Reply:
Other parts of the code make use of `$anchorTimeout`. I think we might indeed be able to have _just_ an `$orphan` flag in future, but to keep this PR smaller I preserved the existing behaviour of this action.

Reply:
The `if (state === 'timeout')` confuses me a little. This wasn't part of the logic of the previous code. It seems that it previously set _both_ `$anchorTimeout` and `$orphan`, but now it sets _either_ one or the other, depending on this new if?

Reply:
Actually this was a bug in the previous code, when the `UPDATE_ANCHOR_STATUS` action was dispatched from this file, the `isOrphan` property was missing from the action, so `$orphan` was always being set to `undefined` here. That happens to be a no-op, but conceptually it is a bug.